src/gui/sudoku_grid.rs

```diff
@@ -6,152 +6,155 @@ use crate::cnf_converter::create_tupples_from_constraints;
-        height = (height - block_size * 3.0) / 2.0;
-        width = width + (width - block_size * 3.0) / 2.0;
-
-        let mut top_left = Pos2::new(width, height);
-        let mut bottom_right = top_left + Vec2::new(cell_size, cell_size);
-
-        let mut draw_constraints = false;
-        let mut constraints: Vec<(i32, i32, i32)> = Vec::new();
-
-        if let Some(num) = app.filter.clicked_constraint_index {
-            constraints = app.rendered_constraints[num].clone();
-            draw_constraints = true;
-
-            // sort them so don't have to search in loop
-            constraints.sort_by(
-                |(r1, c1, _), (r2, c2, _)| {
-                    if r1 != r2 {
-                        r1.cmp(r2)
-                    } else {
-                        c1.cmp(c2)
-                    }
-                },
-            );
-        }
-
-        let mut c_index = 0;
-
-        // row
-        for (i, row) in app.sudoku.iter().enumerate().take(9) {
-            // block divider
-            if i % 3 == 0 && i != 0 {
-                top_left.y += block_spacing;
-                bottom_right = top_left + Vec2::new(cell_size, cell_size);
+impl SATApp {
+    pub fn sudoku_grid(&mut self, ui: &mut Ui, mut height: f32, mut width: f32) -> Response {
+        ui.horizontal_wrapped(|ui| {
+            let block_spacing = 2.0;
+            let square_spacing = 1.0;
+
+            width += block_spacing;
+            let mut cell_size = cmp::min(height as i32, width as i32) as f32;
+            cell_size /= 9.0;
+
+            let block_size = cell_size * 3.0;
+
+            // using these centers the sudoku in the middle of its column
+            height = (height - block_size * 3.0) / 2.0;
+            width = width + (width - block_size * 3.0) / 2.0;
+
+            let mut top_left = Pos2::new(width, height);
+            let mut bottom_right = top_left + Vec2::new(cell_size, cell_size);
+
+            let mut draw_constraints = false;
+            let mut constraints: Vec<(i32, i32, i32)> = Vec::new();
+
+            if let Some(num) = self.filter.clicked_constraint_index {
+                constraints = self.rendered_constraints[num].clone();
+                draw_constraints = true;
+
+                // sort them so don't have to search in loop
+                constraints.sort_by(
+                    |(r1, c1, _), (r2, c2, _)| {
+                        if r1 != r2 {
+                            r1.cmp(r2)
+                        } else {
+                            c1.cmp(c2)
+                        }
+                    },
+                );
             }
-            // square divider
-            top_left.y += square_spacing;
-            bottom_right.y += square_spacing;
 
-            // column
-            for (ii, val) in row.iter().enumerate().take(9) {
+            let mut c_index = 0;
+
+            // row
+            for (i, row) in self.sudoku.iter().enumerate().take(9) {
```

Comment:
Variable names such as i, j or ii are difficult to follow here. While improving the code quality, these should be renamed.